src/wifi/helper/dmg-wifi-helper.cc

```diff
@@ -191,6 +191,15 @@ DmgWifiHelper::Install (const DmgWifiPhyHelper &phyHelper,
       Ptr<DmgWifiMac> mac = StaticCast<DmgWifiMac> (macHelper.Create ());
       Ptr<DmgWifiPhy> phy = StaticCast<DmgWifiPhy> (phyHelper.Create (node, device));
       Ptr<Codebook> codebook = m_codeBook.Create<Codebook> ();
+      Ptr<DmgApWifiMac> ap = DynamicCast<DmgApWifiMac> (mac); // safe downcast
```

Comment:
i would move it up, right after the definition of the mac variable and call it apMac, instead